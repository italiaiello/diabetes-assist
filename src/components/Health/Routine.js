import React from 'react';

const Routine = props => {

    return (
        <article>
            <div className="routineDay">
                <h2>{props.dayOfWeek}</h2>
                <p>{props.medication}</p>
                <p>{props.medication}</p>
            </div>
        </article>
    )
}

export default Routine;