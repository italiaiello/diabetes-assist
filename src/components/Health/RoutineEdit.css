.editRoutine-modal {
    background-color: rgb(0, 59, 70, 0.7);
    position: fixed;
    height: 100%;
    width: 100%;
    top: 0;
    left: 0;
    display: flex;
    align-items: center;
    justify-content: center;
}

.modal-container {
    width: 70%;
    height: 500px;
    display: flex;
    justify-content: center;
    align-items: center;
    flex-direction: column;
    background-color: rgb(102, 165, 173, 0.9);
    border-radius: 50px;
}

.modal-heading {
    display: flex;
    justify-content: center;
    align-items: center;
    position: relative;
    width: 100%;
    height: 80px;
    color: #003B46;
}

.currentRoutine {
    display: flex;
    justify-content: center;
    align-items: center;
    text-align: center;
}

.modal-close {
    position: absolute;
    right: 50px;
    width: 40px;
    height: 40px;
    border-radius: 50%;
    font-size: 25px;
    background-color: #003B46;
    color: white;
    border: none;
    transition: all .2s ease-in-out;
    cursor: pointer;
    outline: none;
}

.modal-close:hover {
    transform: scale(1.05, 1.05)
}

.modal-close:active {
    transform: scale(0.95, 0.95)
}

.modal-buttons {
    margin-top: 30px;
    display: flex;
    justify-content: space-evenly;
    width: 50%;
}

.modal-buttons button {
    width: 200px;
    height: 60px;
    border-radius: 40px;
    outline: none;
    font-size: 15px;
    font-weight: bold;
    transition: all .2s ease-in-out;
    cursor: pointer;
}

.modal-buttons button:hover {
    transform: scale(1.05, 1.05)
}

.modal-buttons button:active {
    transform: scale(0.95, 0.95)
}

.modal-save {
    background-color: #003B46;
    border: none;
    color: white;
}

.modal-cancel {
    border: 2px solid#003B46;
    background: none;
    color: #003B46;
}