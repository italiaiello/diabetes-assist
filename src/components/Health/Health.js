import React, { useState } from 'react'
import HomeButton from '../../images/Home.svg'
import Bell from '../../images/notifications.svg'
import Person from '../../images/Person.png'
import Routine from '../Health/Routine'
import Modal from '../Modal/Modal'
import RoutineEdit from './RoutineEdit'

const Health = props => {

    let daysOfWeek = ['SUN', 'MON', 'TUE', 'WED', 'THU', 'FRI', 'SAT']

    const [isProfileOpen, setIsProfileOpen] = useState(false)

    const toggleModal = () => {
        console.log(isProfileOpen)
        setIsProfileOpen(!isProfileOpen)
    }

    return (
        <article className="pageDisplay">
            <div className="backAndHeading">
                <div className="sectionHeading">
                    <div className="back">
                        <div className="leftArrow" onClick={props.onRouteChange.bind(this, 'home')}></div>
                        <p>Back</p>
                    </div>
                    <h1>My Health</h1>
                </div>
            </div>
            {
                isProfileOpen &&
                <Modal>
                    <RoutineEdit setIsProfileOpen={setIsProfileOpen} toggleModal={toggleModal}/>
                </Modal>
            }
            <article id="healthSection">
                <article id="healthData">
                    <article id="routine">
                        <div id="routineHeader">
                            <h2>Medication Routine</h2>
                            <figure>
                                <img src={Bell} alt="Notifications bell" />
                            </figure>
                            <button id="change" onClick={toggleModal}>Change</button>
                        </div>
                        <div id="routineContent">
                            {
                                daysOfWeek.map((day, i) => {
                                    return (
                                        <Routine 
                                            key={i}
                                            dayOfWeek={daysOfWeek[i]}
                                            medication="Metformin"
                                        />
                                    )
                                })
                            }
                        </div>
                    </article>
                    <article id="bglGraph">
                        <h3>Blood Glucose Level History</h3>
                    </article>
                </article>
                <article id="account">
                    <figure>
                        <img src={Person} alt="Person" />
                    </figure>
                    <p>Aysha<br/>Whitmore</p>

                </article>
            </article>
            <figure className="homeButton">
                <img src={HomeButton} alt="Home button" onClick={props.onRouteChange.bind(this, 'home')} />
            </figure>
        </article>
    )
}

export default Health