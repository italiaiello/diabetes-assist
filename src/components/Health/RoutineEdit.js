import React from 'react'
import './RoutineEdit.css'

const RoutineEdit = ({ isProfileOpen, toggleModal }) => {
    return <div className="editRoutine-modal">
            <button onClick={toggleModal}>Click</button>
        </div>
}

export default RoutineEdit