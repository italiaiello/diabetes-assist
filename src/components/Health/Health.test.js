import React from 'react';
import { shallow } from 'enzyme';
import Health from './Health'

describe('Health.js', () => {
    it('expects to render Health component', () => {
      expect(shallow(<Health />)).toMatchSnapshot()
    })
  }) 