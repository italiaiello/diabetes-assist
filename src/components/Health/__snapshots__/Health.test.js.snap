// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Health.js expects to render Health component 1`] = `ShallowWrapper {}`;
